bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicator.java

```diff
@@ -169,15 +171,15 @@ public synchronized void start(Configuration configuration, MetricsSource metric
         configuration, metricsSource, connection).createIncompatibleMutationAdapter();
   }
 
-  protected void stop() {
+  public void stop() {
 
     LOG.error("Stopping replication to CBT for this EndPoint. ",
         new RuntimeException("Dummy exception for stacktrace"));
 
     // Connection is shared by all the instances of this class, close it only if no one is using it.
     // Closing the connection is required as it owns the underlying gRpc connections and gRpc does
     // not like JVM shutting without closing the gRpc connections.
-    synchronized (HbaseToCloudBigtableReplicationEndpoint.class) {
+    synchronized (this) {
```

Comment:
This was synchronized on the class as its locking static variables. Making it lock on the object may lead to race conditions.

Reply:
synchronized on the class